linq4j/src/main/java/org/apache/calcite/linq4j/EnumerableDefaults.java

```diff
@@ -1113,7 +1121,14 @@ public boolean moveNext() {
               final TSource outer = outers.current();
               final Enumerable<TInner> innerEnumerable;
               if (outer == null) {
-                innerEnumerable = null;
+                if (isConditionAlwaysTrue) {
+                  // when the join condition is always true, the innerLookup will always
+                  // have one entry whose key is an EmptyList and value is
+                  // the whole inner enumerable.
+                  innerEnumerable = innerLookup.values().iterator().next();
```

Comment:
Do we need to add Assertion here for double check?

Reply:
I think we don't need to, cause a always true condition will drain all the inner enumerables definitely into one entry